modules/oadp-pod-crash-set-resource-request-velero.adoc

```diff
@@ -0,0 +1,31 @@
+// Module included in the following assemblies:
+//
+// * backup_and_restore/application_backup_and_restore/troubleshooting.adoc
+
+:_content-type: PROCEDURE
+[id="oadp-pod-crash-resource-request-velero_{context}"]
+= Setting resource requests for a Velero pod
+
+You can use the `configuration.velero.podConfig.resourceAllocations` specification field in the `oadp_v1alpha1_dpa.yaml` file to set specific resource requests for a `Velero` pod.
+
+.Procedure
+
+* Set the `cpu` and `memory` resource requests in the `yaml` file:
+
+.Example Velero file
+
+[source,yaml]
+----
+apiVersion: oadp.openshift.io/v1alpha1
+kind: DataProtectionApplication
+...
```

Comment:
@RichardHoch, why are there three ellipsis here? The API is here: https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#DataProtectionApplication

So I'd say that means this is supposed to be:

```
apiVersion: oadp.openshift.io/v1alpha1
kind: DataProtectionApplication
metadata:
  name: DO WE NEED TO SPECIFY THIS?
  namespace: WHAT NAMESPACE DOES SOMEONE USE?
spec:
  configuration:
    (the rest...)
```

The namespace _might_ be `openshift-adp`, but you'd have to ask one of the developers. And you can't create an object without specifying a `name` as far as I know, although perhaps you can hand waive that away with `...` such as like:

```
metadata:
   ...
```

But that's not optimal because it leaves unanswered questions.

Relevant API parts:

```
type DataProtectionApplication struct {
	[metav1](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1).[TypeMeta](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#TypeMeta)   `json:",inline"`
	[metav1](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1).[ObjectMeta](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta) `json:"metadata,omitempty"`

	Spec   [DataProtectionApplicationSpec](https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#DataProtectionApplicationSpec)   `json:"spec,omitempty"`
	Status [DataProtectionApplicationStatus](https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#DataProtectionApplicationStatus) `json:"status,omitempty"`
}

type DataProtectionApplicationSpec struct {
	// BackupLocations defines the list of desired configuration to use for BackupStorageLocations
	// +optional
	[BackupLocation](https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#BackupLocation)s []BackupLocation `json:"backupLocations"`
	// VolumeSnapshotLocations defines the list of desired configuration to use for VolumeSnapshotLocations
	// +optional
	[SnapshotLocation](https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#SnapshotLocation)s []SnapshotLocation `json:"snapshotLocations"`
	// UnsupportedOverrides can be used to override the deployed dependent images for development
	// +optional
	UnsupportedOverrides map[[UnsupportedImageKey](https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#UnsupportedImageKey)][string](https://pkg.go.dev/builtin#string) `json:"unsupportedOverrides,omitempty"`
	// add annotations to pods deployed by operator
	// +optional
	PodAnnotations map[[string](https://pkg.go.dev/builtin#string)][string](https://pkg.go.dev/builtin#string) `json:"podAnnotations,omitempty"`
	// DNSPolicy defines how a pod's DNS will be configured.
	// https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy
	// +optional
	PodDnsPolicy [corev1](https://pkg.go.dev/k8s.io/api/core/v1).[DNSPolicy](https://pkg.go.dev/k8s.io/api/core/v1#DNSPolicy) `json:"podDnsPolicy,omitempty"`
	// PodDNSConfig defines the DNS parameters of a pod in addition to
	// those generated from DNSPolicy.
	// https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config
	// +optional
	PodDnsConfig [corev1](https://pkg.go.dev/k8s.io/api/core/v1).[PodDNSConfig](https://pkg.go.dev/k8s.io/api/core/v1#PodDNSConfig) `json:"podDnsConfig,omitempty"`
	// BackupImages is used to specify whether you want to deploy a registry for enabling backup and restore of images
	// +optional
	BackupImages *[bool](https://pkg.go.dev/builtin#bool) `json:"backupImages,omitempty"`

	// Configuration is used to configure the data protection application's server config
	Configuration *[ApplicationConfig](https://pkg.go.dev/github.com/openshift/oadp-operator/api/v1alpha1#ApplicationConfig) `json:"configuration"`
}
```